File: server/router/ExpertDatabase/expert_achievement.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertAchievementRouter struct{}

// InitExpertAchievementRouter 初始化 专家成果 路由信息
func (s *ExpertAchievementRouter) InitExpertAchievementRouter(Router *gin.RouterGroup) {
	expertAchievementRouter := Router.Group("expertAchievement").Use(middleware.OperationRecord())
	expertAchievementRouterWithoutRecord := Router.Group("expertAchievement")
	expertAchievementApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertAchievementApi
	{
		expertAchievementRouter.POST("createExpertAchievement", expertAchievementApi.CreateExpertAchievement)             // 新建专家成果
		expertAchievementRouter.DELETE("deleteExpertAchievement", expertAchievementApi.DeleteExpertAchievement)           // 删除专家成果
		expertAchievementRouter.DELETE("deleteExpertAchievementByIds", expertAchievementApi.DeleteExpertAchievementByIds) // 批量删除专家成果
		expertAchievementRouter.PUT("updateExpertAchievement", expertAchievementApi.UpdateExpertAchievement)              // 更新专家成果
	}
	{
		expertAchievementRouterWithoutRecord.GET("findExpertAchievement", expertAchievementApi.FindExpertAchievement)       // 根据ID获取专家成果
		expertAchievementRouterWithoutRecord.GET("getExpertAchievementList", expertAchievementApi.GetExpertAchievementList) // 分页获取专家成果列表
	}
}

File: web/src/api/expertAchievement.js
import service from '@/utils/request'

// @Tags ExpertAchievement
// @Summary 创建专家成果
// @Router /expertAchievement/createExpertAchievement [post]
export const createExpertAchievement = (data) => {
  return service({
    url: '/expertAchievement/createExpertAchievement',
    method: 'post',
    data
  })
}

// @Tags ExpertAchievement
// @Summary 删除专家成果
// @Router /expertAchievement/deleteExpertAchievement [delete]
export const deleteExpertAchievement = (params) => {
  return service({
    url: '/expertAchievement/deleteExpertAchievement',
    method: 'delete',
    params
  })
}

// @Tags ExpertAchievement
// @Summary 批量删除专家成果
// @Router /expertAchievement/deleteExpertAchievementByIds [delete]
export const deleteExpertAchievementByIds = (params) => {
  return service({
    url: '/expertAchievement/deleteExpertAchievementByIds',
    method: 'delete',
    params
  })
}

// @Tags ExpertAchievement
// @Summary 更新专家成果
// @Router /expertAchievement/updateExpertAchievement [put]
export const updateExpertAchievement = (data) => {
  return service({
    url: '/expertAchievement/updateExpertAchievement',
    method: 'put',
    data
  })
}

// @Tags ExpertAchievement
// @Summary 用id查询专家成果
// @Router /expertAchievement/findExpertAchievement [get]
export const findExpertAchievement = (params) => {
  return service({
    url: '/expertAchievement/findExpertAchievement',
    method: 'get',
    params
  })
}

// @Tags ExpertAchievement
// @Summary 分页获取专家成果列表
// @Router /expertAchievement/getExpertAchievementList [get]
export const getExpertAchievementList = (params) => {
  return service({
    url: '/expertAchievement/getExpertAchievementList',
    method: 'get',
    params
  })
}
